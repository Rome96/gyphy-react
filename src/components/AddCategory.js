import React, { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = props => {
  const { setCategories } = props
  const [value, setValue] = useState("");

  const handleInputChange = (e) => {
    const texInput = e.target.value;
    setValue(texInput);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (value.trim().length > 0) {
      setCategories( cats => [value,...cats])
      setValue('')
    }
  };


  return (
    <form onSubmit={handleSubmit}>
      <input
        type="value"
        placeholder="Search..."
        value={value}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory
