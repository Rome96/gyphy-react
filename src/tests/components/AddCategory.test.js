import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe("Testing in components <AddCategory/>", () => {
  const setCategories = () => {};

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  
  test("Render component <AddCategory />", () => {
    expect(wrapper).toMatchSnapshot()
  });

  test('onChange input search', () => {
    const input = wrapper.find('input')
    const value = 'Hello World'
    input.simulate('change', { target: { value } })

  });

});
