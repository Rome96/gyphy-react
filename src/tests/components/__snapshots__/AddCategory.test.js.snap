// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in components <AddCategory/> Render component <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="value"
    value=""
  />
</form>
`;
